tests/unittests/test_retry_logic.py

```diff
@@ -0,0 +1,74 @@
+import unittest
+from unittest.mock import patch
+import time
+from requests.models import Response
+from tap_outreach import client
+
+class Mockresponse(Response):
+    def __init__(self, status_code):
+        super().__init__()
+        self.status_code = status_code
+        self.headers = {"x-ratelimit-reset": time.time()}
+    
+    def raise_for_status(self):
+        pass
+    
+class TestRetryLogic(unittest.TestCase):
```

Comment:
Add an unittest for non retriable http error code as well.

Reply:
done